src/main/java/com/scrive/pdftools/tools/LambdaInterface.java

```diff
@@ -0,0 +1,135 @@
+            getFileContent(s3fileName, outputStream).ifPresent(fileContent -> {
+                final JSONObject jsonObject;
+                try {
+                    jsonObject = new JSONObject(fileContent);
+                }catch (Exception e) {
+                    e.printStackTrace();
+                    LambdaInterface.respondToOutputStreamWithErrorResponse(outputStream, "Can't parse spec as JSON");
+                    return;
+                }
+
+                try {
+                    System.out.println("DEBUG: Started flattening action on document");
+                    ByteArrayOutputStream out1 = FlattenerHandler.execute(jsonObject);
+                    String resultFileName = RESP_FILE_NAME_PREFIX + s3fileName;
+                    S3ForLambda.putStringToAmazonFile(resultFileName, out1.toByteArray());
+                    JSONObject response = new JSONObject();
+                    response.put(RESP_RESULT_S3_FILE_NAME, resultFileName);
+                    response.put(RESP_SUCCESS, true);
+                    respondToOutputStreamWith(outputStream, response);
+                    System.out.println("DEBUG: Finished flattening action on document");
+                }catch(Exception e) {
+                    e.printStackTrace();
+                    respondToOutputStreamWithErrorResponse(outputStream, e.getMessage());
+                }
+            });
+        });
+    }
+
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) {
+
+        JSONObject callJSON;
+        try {
+            String fs = S3ForLambda.fromStreamToString(inputStream);
+            JSONObject inputJSON = new JSONObject(fs);
+            boolean hasRequestBodyHiddenInJSON = inputJSON.has(RQ_PARAM_ACTION_BODY);
+            if (hasRequestBodyHiddenInJSON) {
+                callJSON = new JSONObject(inputJSON.getString(RQ_PARAM_ACTION_BODY));
+            } else {
+                callJSON = inputJSON;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            LambdaInterface.respondToOutputStreamWithErrorResponse(outputStream,"Invalid input - call spec has to be a json");
+            return;
+        }
+
+        if (!callJSON.has(RQ_PARAM_ACTION)) {
+            LambdaInterface.respondToOutputStreamWithErrorResponse(outputStream,"Invalid action. Action has to be set");
+            return;
+        }
+
+        switch (callJSON.getString(RQ_PARAM_ACTION)) {
+            case RQ_PARAM_ACTION_FLATTEN:
+                System.out.println("DEBUG: Processing flatten action");
+                LambdaInterface.handleFlattenAction(callJSON, outputStream);
+                return;
+            default:
+                System.out.println(String.format("ERROR: Action %s isn't permitted", callJSON.getString(RQ_PARAM_ACTION)));
+        }
+    }
+
+    private static void respondToOutputStreamWith(OutputStream outputStream, JSONObject response) {
```

Comment:
maybe `respondWith` and below `respondWithError` would be more pithy names

Reply:
Yes. 👍

Reply:
Yes. Renamed